src/EnergyPlus/ScheduleManager.cc

```diff
@@ -1641,354 +1636,229 @@ namespace ScheduleManager {
-                                         "ProcessScheduleInput: For " + CurrentModuleObject + "=\"" + Alphas(1) + "\", " + cAlphaFields(2) + "=\"" +
-                                             Alphas(2) + "\" not found -- will not be validated");
-                    } else {
-                        ShowWarningError(state,
-                                         "For " + CurrentModuleObject + "=\"" + Alphas(1) + "\", Blank " + cAlphaFields(2) +
-                                             " input -- will not be validated.");
-                    }
-                } else {
-                    state.dataScheduleMgr->Schedule(SchNum).ScheduleTypePtr = CheckIndex;
+
+        // Runs getObjectItem for Schedule:File and saves it to vector in state: allIdfSchedData
+        PreProcessIDF(state, SchNum, NumCommaFileSchedules);
+
+        // Because we are focusing on saving performance by decreasing number of file open-close operations, get set of filenames
+        std::set<std::string> setOfFilenames;
+        PopulateSetOfFilenames(state.dataScheduleMgr->allIdfSchedData, setOfFilenames);
+
+        // we create a new vector of data with the pre-processed data, arranged according to filenames
+        // we use fileName in the outer loop so as to minimize opening and closing files
+        for (const std::string &fileNameItem : setOfFilenames) {
+            for (schedInputIdfPreprocessObject &item : state.dataScheduleMgr->allIdfSchedData) {
+                if (item.fileName == fileNameItem) {
+                    int numberOfRows = (item.numHourlyValues * 60.0) / item.MinutesPerItem;
+                    std::vector<Real64> vals;
+                    vals.reserve(numberOfRows);                                            // pre-allocating the big vector
+                    item.setColumnarDataIndex(state.dataScheduleMgr->columnarData.size()); // TODO: this is a kludgy-hack
+
+                    // give each schedule it's own column in columnarData
+                    state.dataScheduleMgr->columnarData.emplace_back(item.name,
+                                                                     item.columnOfInterest,
+                                                                     item.numHourlyValues,
+                                                                     item.MinutesPerItem,
+                                                                     item.rowsToSkip,
+                                                                     item.FileIntervalInterpolated,
+                                                                     item.fileName,
+                                                                     item.columnSeperator,
+                                                                     item.ScheduleTypePtr,
+                                                                     item.SchNum,
+                                                                     vals);
```

Comment:
The main purpose of this code is 

1. To pre-allocate the very large vectors in columnarData using the reserve method. Calling reserve earlier in the code to improve performance. The other option would have been to allocate the vals vector using push back, which decreases performance. 

2. To connect allIdfSchedData and columnarData: using the setColumnarDataIndex(). This index is used later to create a map between the column of interest in the CSV and the element index of columnarData. 

The reason for this: so I could iterate through the CSV files while knowing which column of the CSV file corresponds to which column of columnarData.